runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -161,7 +111,7 @@ end
 ---@param bufnr (integer) Buffer handle, or 0 for current
 ---@private
 local function clear(bufnr)
-  bufnr = resolve_bufnr(bufnr)
+  bufnr = bufnr or api.nvim_get_current_buf()
```

Comment:
These changes also change semantics as they don't handle the bufnr = 0 case. I suspect that's not intentional?

Reply:
Ah, sorry, I didn't realize that 0 is truthy in lua. This definitely changes what was intended here.